examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/provisioning/DeviceProvisioningFragment.kt

```diff
@@ -87,8 +86,40 @@ class DeviceProvisioningFragment : Fragment() {
     val port = 5540
     val id = DeviceIdUtil.getNextAvailableId(requireContext())
     val deviceController = ChipClient.getDeviceController(requireContext())
+
     DeviceIdUtil.setNextAvailableId(requireContext(), id + 1)
     deviceController.setCompletionListener(ConnectionCallback())
+
+    deviceController.setDeviceAttestationDelegate(DEVICE_ATTESTATION_FAILED_TIMEOUT,
+    { devicePtr, attestationInfo, errorCode ->
+      Log.i(TAG, "Device attestation errorCode: $errorCode, " +
+              "Look at 'src/credentials/attestation_verifier/DeviceAttestationVerifier.h' " +
+              "AttestationVerificationResult enum to understand the errors")
+
+      if (errorCode != STATUS_PAIRING_SUCCESS) {
+        requireActivity().runOnUiThread(Runnable {
+          val alertDialog: AlertDialog? = activity?.let {
```

Comment:
Since you're already using `requireActivity()`, you can probably refactor this a bit to avoid the potential null handling:

```
val activity = requireActivity()
val dialog = 
  AlertDialog.Builder(activity)
    .setPositiveButton("Continue") { deviceController.continueCommissioning(devicePtr, true) }
    .setNegativeButton("No") { deviceController.continueCommissioning(devicePtr, false) }
    .setTitle("Device Attestation")
    .setMessage("Device Attestation failed for device under commissioning. Do you wish to continue pairing?")
    .create()

activity.runOnUiThread { dialog.show() }
```